SDK_dotnet/Clients/GluwaClient.cs

```diff
@@ -374,24 +418,116 @@ private string getContractAddress(ECurrency currency)
+            string encodedData = Convert.ToBase64String(signatureByte);
+
+            return encodedData;
+        }
+
+        private string getGluwacoinTransactionSignature(ECurrency currency, string amount, string fee, string nonce, string address, string target, string privateKey)
+        {
+            BigInteger convertAmount = GluwacoinConverter.ConvertToGluwacoinBigInteger(amount);
+            BigInteger convertFee = GluwacoinConverter.ConvertToGluwacoinBigInteger(fee.ToString());
+
+            ABIEncode abiEncode = new ABIEncode();
+            byte[] messageHash = abiEncode.GetSha3ABIEncodedPacked(
+                new ABIValue("address", getContractAddress(currency)),
+                new ABIValue("address", address),
+                new ABIValue("address", target),
+                new ABIValue("uint256", convertAmount),
+                new ABIValue("uint256", convertFee),
+                new ABIValue("uint256", int.Parse(nonce))
+                );
+
+            EthereumMessageSigner signer = new EthereumMessageSigner();
+            string signature = signer.Sign(messageHash, privateKey);
+
+            return signature;
+        }
+
+        private async Task<string> getBtcTransactionSignatureAsync(ECurrency currency, string address, string amount, string fee, string target, string privateKey)
+        {
+            Result<BalanceResponse, ErrorResponse> getUnspentOutput = await GetBalanceAsync(currency, address, true);
+            List<UnspentOutput> unspentOutputs = getUnspentOutput.Data.UnspentOutputs.OrderByDescending(u => u.Amount).ToList();
+
+            Money amountValue = Money.Parse(amount);
+            Money feeValue = Money.Parse(fee);
+            Money totalAmountAndFeeValue = amountValue + feeValue;
+            BigInteger totalAmountAndFee = new BigInteger(totalAmountAndFeeValue.Satoshi);
+
+            BitcoinAddress sourceAddress = BitcoinAddress.Create(address, mNetwork);
+            BitcoinAddress targetAddress = BitcoinAddress.Create(target, mNetwork);
+            BitcoinSecret secret = new BitcoinSecret(privateKey, mNetwork);
+
+            List<UnspentOutput> usingUnspentOutputs = new List<UnspentOutput>();
+            BigInteger unspentOutputTotalAmount = BigInteger.Zero;
+            for (int i = 0; i < unspentOutputs.Count; i++)
+            {
+                if (unspentOutputTotalAmount >= totalAmountAndFee || i >= MAX_UNSPENTOUTPUTS_COUNT)
+                {
+                    break;
+                }
+
+                usingUnspentOutputs.Add(unspentOutputs[i]);
+                Money sumAmount = Money.Parse(unspentOutputs[i].Amount);
+                unspentOutputTotalAmount += new BigInteger(sumAmount.Satoshi);
+            }
+
+            if (usingUnspentOutputs.Count >= MAX_UNSPENTOUTPUTS_COUNT)
+            {
+                throw new Exception($"Could not find up to {MAX_UNSPENTOUTPUTS_COUNT} BTC unspent outputs that can cover the amount and fee.");
```

Comment:
Reminder: another generic exception here. Resolve if we decide not to change it

Reply:
Use InvalidOperationException instead